internal/controller/driver_controller.go

```diff
@@ -477,6 +479,9 @@ func (r *driverReconcile) reconcileControllerPluginDeployment() error {
 										utils.CsiAddonsEndpointContainerArg,
 										"",
 									),
+									utils.If(r.logRotation(), utils.LogRotationSpecArg(r.driver.Spec.Log.Rotation.LogHostPath, deploy.Name, fmt.Sprintf("csi-%splugin", r.driverType)), ""),
```

Comment:
@parth-gr 
This will panic, unlike a built-in `if` all the arguments are first evaluated and only then passed into the `util. If` function (this is the normal behavior for any function call in most languages). This means that the expression `r.driver.Spec.Log.Rotation.LogHostPath` will try to be evaluated no matter what the result of `r.logRotation()` is. 
This in turn will result in a nil de-reference which will panic.